robot_calibration/src/ground_plane_finder.cpp

```diff
@@ -131,16 +131,46 @@ bool GroundPlaneFinder::find(robot_calibration_msgs::CalibrationData * msg)
   sensor_msgs::PointCloud2Iterator<float> iter_cloud(cloud_, "x");
 
   // Set msg size
-  msg->observations.resize(2);
-  msg->observations[0].sensor_name = camera_sensor_name_;
-  msg->observations[0].features.resize(points_total);
-  msg->observations[1].sensor_name = chain_sensor_name_;
-  msg->observations[1].features.resize(points_total);
+  int idx_cam = -1;
+  int idx_chain = -1;
 
-  size_t step = cloud_.width/points_total;
+  if (msg->observations.size() == 0)
```

Comment:
we probably don't need this block -- each finder should be the exclusive creator for each sensor, right? thus, we should always just append two new observations